Interfaces/API/Job.py

```diff
@@ -617,6 +615,30 @@ def setDestination(self, destination):
       return self._reportError('Invalid destination site, expected string or list', **kwargs)
     return S_OK()
 
+  #############################################################################
+  def setNumberOfProcessors(self, processors):
+    """Helper function.
+
+       Example usage:
+
+       >>> job = Job()
+       >>> job.setNumberOfProcessors(4)
+
+       :param processors: number of processors required by the job
+       :type processors: Int
+    """
+    kwargs = {'processors': processors}
+    if not isinstance(processors, int):
+      try:
+        processors = int(processors)
+      except ValueError:
+        if not re.search('{{', processors):
```

Comment:
What is the use-case when "{{" is in processors? (`if '{{' in processors:` should work as well, right? '{{' is not a regular expression, is it?)
Also if "{{" is in `processors` the `'%d' % processors` will raise an exception

Reply:
Honestly, I don't know. I copy-pasted this from `setCPUTime`, so I guess there is (or there was?) a reason for that, but I ignore it. Maybe @atsareg can shed some light.

Reply:
in `setCPUTime` the parameter is added as a workflow parameter, so maybe that does something with the "{{". 
But here it would just lead to an exception.

Reply:
https://github.com/DIRACGrid/DIRAC/commit/e8deb898fc556ad3c620882438c51872c8ca84a7
The "{{" is (was?) for workflow templates

Reply:
Removed.